pkgs/by-name/tr/trigger-control/package.nix

```diff
@@ -16,15 +17,17 @@ let
   inherit (gnome) zenity;
 in
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "trigger-control";
-  version = "unstable-2023-06-18";
+  version = "1.5.1";
 
   src = fetchFromGitHub {
     owner = "Etaash-mathamsetty";
     repo = "trigger-control";
-    rev = "d457ebd9e0844cfc456bfa4fa4bb694bb8ad982a";
-    hash = "sha256-QWhUQ8xqS8oRVF0KUpEthlrOoXmhcfEkIHauDI1/5a8=";
+    # upstream does not use consistant tags pattern, so we use git commit hash
+    # https://github.com/Etaash-mathamsetty/trigger-control/tags
+    rev = "7b46e743227830d3a97720067d0a6cf20133af90";
+    hash = "sha256-wueNjrCPWDw7iCOuXgzWZiweRc3o6SUw3xZGgm078nM=";
```

Comment:
I am getting a hash failure -- setting `hash = "";` then running build can get the new hash which is ...
```
hash = "sha256-nWSvsgksZ4Cxy1+i0xy8pNalgsiAuaqxNVwT/CThaBI=";
```

Reply:
Fixed.

My bad. I way using `rev = "v.1.5.1"` to test the build, and then switched to `rev = "7b46e743227830d3a97720067d0a6cf20133af90"`, without noticing that `fetchFromGitHub` uses different ways to get the source code by tags and commit hash.

Reply:
7b46e743227830d3a97720067d0a6cf20133af90 is not the commit referenced by the "v.1.5.1" tag, it is the head on main. so you were using a different rev, 011d0c3657e229c1902cde955e71a64b0c7d91b1 aka "v.1.5.1", which should give the same hash.

anyway, 7b46e743227830d3a97720067d0a6cf20133af90 seems fine as that just makes the version in the header file "1.5.1".
https://github.com/Etaash-mathamsetty/trigger-control/compare/v.1.5.1...main